pkg/operator/controllers/workaround/routefix.go

```diff
@@ -137,17 +99,17 @@ func (i *routeFix) CreateOrUpdate(ctx context.Context) error {
 		Name: "privileged-routefix",
 	}
 	scc.Groups = nil
-	scc.Users = []string{kubeServiceAccount}
+	scc.Users = []string{kubeServiceAccountRoutefix}
 
-	_, err = i.seccli.SecurityV1().SecurityContextConstraints().Create(scc)
+	_, err = r.seccli.SecurityV1().SecurityContextConstraints().Create(scc)
 	if err != nil && !errors.IsAlreadyExists(err) {
 		return err
 	}
 
-	return i.applyDaemonSet(&appsv1.DaemonSet{
+	return r.applyDaemonSet(&appsv1.DaemonSet{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      "routefix",
-			Namespace: kubeNamespace,
+			Name:      r.Name(),
```

Comment:
this is changing "routefix" to "routeFix" which is not a compliant k8s name because it has a capital letter.
a) I don't believe this code can work
b) If it did, it could cause an upgrade problem where there were two daemonsets.

Please fix and add a new e2e test which proves that the routefix daemonset runs on a deployed cluster.